Dispatcher/Repositories/ArticlesRepository.swift

```diff
@@ -0,0 +1,91 @@
+            switch result {
+            case .success(let dataResult):
+                for article in dataResult.articles {
+                    if(article.language == "en") {
+                        let article = Article(
+                            id: article.id,
+                            title: article.title,
+                            summary: article.summary,
+                            author: article.author,
+                            topic: [article.topic],
+                            imageURL: article.media,
+                            language: article.language,
+                            date: article.publishedDate)
+                        self.articles.append(article)
+                        completionHandler(self.articles, Int(dataResult.totalPages), nil)
+                    }
+                }
+            case .failure(let error):
+                completionHandler([], 0, error.localizedDescription)
+            }
+        }
+    }
+    
+    private func buildLatestArticlesRequest(_ pageNumber: Int) -> Request {
+        return Request(
+            baseUrl: APIConstants.latestHeadlinesNewscatcherURL,
+            headers: ["x-api-key": APIConstants.APIkey],
+            parameters: ["countries" : "IL", "page_size": APIConstants.ARTICLES_TO_FETCH_AMOUNT, "page" : pageNumber],
+            method: .get)
+    }
+    
+    func getArticlesByKeywords(_ searchKeywords: String, pageNumber: Int, completionHandler: @escaping ( _ articles: [Article], _ totalPages: Int, _ errorMsg: String?) -> Void) {
+        self.articles = []
+        let request = buildSearchArticlesRequest(searchKeywords, pageNumber)
+        
+        alamofireManager.sendRequest(request) { (result: Result<ArticleApiObject, Error>) in
+            switch result {
+            case .success(let dataResult):
+                for article in dataResult.articles {
+                    if(article.language == "en") {
+                        let article = Article(
+                            id: article.id,
+                            title: article.title,
+                            summary: article.summary,
+                            author: article.author,
+                            topic: [article.topic],
+                            imageURL: article.media,
+                            language: article.language,
+                            date: article.publishedDate)
+                        self.articles.append(article)
+                        completionHandler(self.articles, Int(dataResult.totalPages), nil)
+                    }
+                }
+            case .failure(let error):
+                completionHandler([], 0, error.localizedDescription)
+            }
+        }
+    }
+}
+    
+private func buildSearchArticlesRequest(_ searchKeywords: String, _ pageNumber: Int) -> Request {
+    return Request(
+        baseUrl: APIConstants.searchArticlesNewscatcherURL,
```

Comment:
You have 2 methods in this file that build a Request with mostly similar contents (url, headers, method). Better have a dedicated function that does this for you and accepts properties that can be different (e.g. parameters array) only if there are indeed different parameters to pass into it